modules/gdscript/gdscript_parser.cpp

```diff
@@ -2783,12 +2783,11 @@ void GDScriptParser::_parse_block(BlockNode *p_block, bool p_static) {
 
 				tokenizer->advance();
 				int var_line = tokenizer->get_token_line();
+				StringName n = tokenizer->get_token_literal();
 				if (!tokenizer->is_token_literal(0, true)) {
-
-					_set_error("Expected an identifier for the local variable name.");
+					_set_error("Cannot use reserved keyword \"" + String(n) + "\" for the local variable name.");
```

Comment:
This is not always this case at this point. If you type `var =` it won't be a literal, but also won't be a reserved keyword.